backend/src/tasks/LookingGlass.ts

```diff
@@ -0,0 +1,286 @@
+      ref: {
+        type: string;
+        id: number;
+      };
+      name: string;
+      type: string;
+      threatNames: string[];
+      locations: {
+        city: string;
+        country: string;
+        region: string;
+        geoPoint: {
+          long: number;
+          lat: number;
+        };
+        countryName: string;
+        sources: string[];
+        country2Digit: string;
+        lastSeen: number;
+      }[];
+      ipv4: number;
+      asns: number[];
+      threatIds: string[];
+      cidrv4s: string[];
+    };
+    ref: {
+      type: string;
+      right: {
+        type: string;
+        id: string;
+      };
+      left: {
+        type: string;
+        id: string;
+      };
+    };
+  }[];
+}
+
+async function getOrg(organizationId: string | undefined) {
+  await connectToDatabase();
+  if (typeof organizationId === 'string') {
+    const organization = await Organization.findOne(organizationId);
+    if (organization) {
+      return organization;
+    } else {
+      console.log('No organization found');
+      exit;
+    }
+  } else {
+    console.log('No OrganizationId provided');
+    exit;
+  }
+}
+
+async function collectionByWorkspace() {
+  const resource =
+    'https://delta.lookingglasscyber.com/api/v1/workspaces/' +
+    'NCATS POV' +
+    '/collections';
+
+  return (await got(resource, {
+    headers: { Authorization: 'Bearer ' + process.env.LG_API_KEY }
+  }).json()) as LGCollectionsResponse[];
+}
+
+function ValidateIPaddress(ipaddress) {
+  if (/\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(ipaddress)) {
+    return true;
+  }
+  return false;
+}
+
+async function getThreatInfo(collectionID) {
+  const resource = 'https://delta.lookingglasscyber.com/api/graph/query';
+  const modifier = {
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+      ['=', 'left.collectionIds', collectionID]
+    ],
+    fields: ['firstSeen', 'lastSeen', 'sources', 'right', 'left'],
+    limit: 100000,
+    workspaceIds: []
+  };
+  const h1 = {
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer ' + process.env.LG_API_KEY
+  };
+  return (got
+    .post(resource, {
+      json: modifier,
+      headers: h1
+    })
+    .json() as unknown) as LGThreatResponse[];
+}
+
+async function saveAndPullDomains(response, organizationId, scanId, Org) {
+  const domains: Domain[] = [];
+  const data = response;
+  for (const l of data['results']) {
+    if (ValidateIPaddress(l['left']['name'])) {
+      const current_Domain = plainToClass(Domain, {
+        name: l['left']['name'],
+        ip: l['left']['name'],
+        organization: { id: organizationId },
+        fromRootDomain: Org['rootDomains'][0],
+        ipOnly: true,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(current_Domain);
+    } else {
+      const current_Domain = plainToClass(Domain, {
+        name: l['left']['name'],
```

Comment:
Can we change all the bracket notation to dot notation? Generally, it's better in js to use dot notation as it's easier to read and quicker to type. The only case we should use bracket notation is if the property contains special characters.

```suggestion
        name: l.left.name,
```